### PR TITLE
Fix leaving the organization

### PR DESCRIPTION
**Pull request by Marishka17**
### Motivation and context

Users should have the option to leave the organization.

### How has this been tested?

Added REST API tests, updated OPA tests
### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added a description of my changes into the [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
~~- [ ] I have updated the documentation accordingly~~
- [x] I have added tests to cover my changes
~~- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))~~
- [x] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: cvat/apps/iam/rules/memberships.rego
Comment (Marishka17) [inline cvat/apps/iam/rules/memberships.rego:98]: @kirill-sizov, In case of deleting a membership, the environment will never be a sandbox. https://github.com/opencv/cvat/blob/9148234a159c5d5e193e1fcd785ed7010bf950b7/cvat/apps/iam/permissions.py#L61C29-L61C29

**Review by kirill-sizov — APPROVED** [review]
LGTM, but if tests are not supposed to be in this PR need to create an issue for it just to not forget about it

File: cvat/apps/iam/rules/memberships.rego
Comment (nmanovic) [inline cvat/apps/iam/rules/memberships.rego:106]: @Marishka17 , why cannot maintainer leave the organization?

Comment (Marishka17) [reply cvat/apps/iam/rules/memberships.rego:106]: @nmanovic, For me, maintainer should also be able to leave the organization. But we have [here](https://github.com/opencv/cvat/blob/783c414cfd08b7d0bc1510936bca955fa0c5a733/cvat/apps/iam/rules/memberships.rego#L80) and [here](https://github.com/opencv/cvat/blob/783c414cfd08b7d0bc1510936bca955fa0c5a733/cvat/apps/iam/rules/tests/configs/memberships.csv?plain=1#L9) such limitations..

Comment (nmanovic) [reply cvat/apps/iam/rules/memberships.rego:106]: The idea behind the rule was a maintainer cannot delete another maintainer. Maintainers should be possible to leave the organization for sure.

**Comment by Marishka17** [comment]
@nmanovic, Could you please take a look?

File: cvat-ui/src/actions/organization-actions.ts
Comment (nmanovic) [inline cvat-ui/src/actions/organization-actions.ts:223]: Can we avoid that?

File: cvat-core/src/organization.ts
Comment (nmanovic) [inline cvat-core/src/organization.ts:335]: Why do we need the change?

**Comment by nmanovic** [comment]
@klakhov , could you please look at the PR? I don't like window.location.reload(). Can we avoid it?

File: cvat-ui/src/actions/organization-actions.ts
Comment (klakhov) [reply cvat-ui/src/actions/organization-actions.ts:223]: I dont think we can avoid reload, we already do it when we change organization to refresh all data in the global state. But now we have `window.location.reload();` in two places related to organizations. I belive this code could be improved with separate action `organizationActions.changeOrganization()` so its all done in one place

File: cvat-core/src/organization.ts
Comment (klakhov) [reply cvat-core/src/organization.ts:335]: I think  we need this change because of [this line](https://github.com/opencv/cvat/blob/develop/cvat/apps/organizations/views.py#L64). Searching by id isnt working, but searching by username works fine.

**Review by nmanovic — APPROVED** [review]

**Pull request merged**
